pkg/cmd/gpg-key/add/http.go

```diff
@@ -21,6 +21,10 @@ func gpgKeyUpload(httpClient *http.Client, hostname string, keyFile io.Reader) e
 		return err
 	}
 
+	if !isGpgKeyArmored(keyBytes) {
+		return errors.New("only ASCII-armored GPG keys are supported")
```

Comment:
Let's not perform client-side validation here because in the future the server might accept binary keys in addition to ASCII-armored ones. Instead, let's react to the server validation message about invalid keys and improve the error message:
- If the server responds with "key invalid" error, scan the key to check whether it's ASCII-armored, and if not, warn the user;
- Consider printing the gpg command-line instructions to convert a gpg key to ASCII-armored format. Something like: `gpg --enarmor <key-file> | gh gpg-key add -`

What do you think?

Reply:
> Let's not perform client-side validation here because in the future the server might accept binary keys in addition to ASCII-armored ones. Instead, let's react to the server validation message about invalid keys and improve the error message:
> 
> * If the server responds with "key invalid" error, scan the key to check whether it's ASCII-armored, and if not, warn the user;
> * Consider printing the gpg command-line instructions to convert a gpg key to ASCII-armored format. Something like: `gpg --enarmor <key-file> | gh gpg-key add -`
> 
> What do you think?

I agree that.

But `--enarmor` is  designed to armor arbitrary input and its armor header(`-----BEGIN PGP ARMORED FILE-----`) is not accepted by server side.  

So I suggest to do it like this :

1. use `gpg --list-keys` to list all public keys on the machine now
2. use `gpg --armor --export <GPG key ID> | gh gpg-key-add -` 

It's better to give a GPG key ID **explicitly**. If two or more public keys are on this machine ,` gpg` will armor all of them and put it in one single file. , the output will be rejected by GitHub.

So the  output should be like this :

```bash
$ gh gpg-key add binary-key-file
it seems that the GPG key is not armored.
please try to find your GPG key ID using:
	gpg --list-keys 
and use command below to add it to your accont:
	gpg --armor --export <GPG key ID> | gh gpg-key-add -
```

How do you think about that?

Reply:
I've done that and force pushed it first. Hope that would be OK.